twine/repository.py

```diff
@@ -121,9 +121,14 @@ def _upload(self, package):
                 (package.basefilename, fp, "application/octet-stream"),
             ))
             encoder = MultipartEncoder(data_to_send)
-            bar = ProgressBar(expected_size=encoder.len, filled_char='=')
+            bar = ProgressBar(total=encoder.len, unit='bytes', unit_scale=True, leave=False)
+            def update_progressbar(monitor, _cache=[0]):
```

Comment:
This relies on a little Python quirk that isn't as intuitive. Thoughts on factoring this out into it's own object and then passing a method of that object to `MultipartEncoderMonitor`?